emacs/radian.el

```diff
@@ -539,13 +578,31 @@ binding the variable dynamically over the entire init-file."
 ;; https://github.com/jwiegley/use-package#notes-about-lazy-loading.
 (setq use-package-always-defer t)
 
+(defmacro radian-use-package (name &rest args)
+  "Like `use-package', but enabled only if the package is not in
+`radian-exclude-packages'. NAME and ARGS are as in `use-package'."
+  (declare (indent 1))
+  `(if (radian-enabled-p ,name)
+       (use-package ,name ,@args)
+     (let* ((melpa-recipe (plist-get ',args ':straight))
+            (non-nil-straight (not (and (plist-member ',args ':straight)
+                                        (null melpa-recipe)))))
+       (when (or (and straight-use-package-by-default
+                      non-nil-straight)
+                 non-nil-straight)
```

Comment:
This is not strictly necessary since `radian-use-package` is never called with `:straight nil` and `straight-use-package-by-default` is set to `t`. But I thought it's better to make it more robust.

Reply:
It might be better to simply document the limitation, because there are some additional complexities with actually parsing out `:straight`. In particular, `use-package` arguments aren't actually a plist---for example, `:init` and `:config` can take multiple forms. The arguments get parsed and turned into a proper plist internally in `use-package`. But in principle the code as written could fail if someone put the arguments in a different order.

Given that we have `use-feature` (maybe should be aliased to `radian-use-feature`?) to serve the case of `:straight nil`, it might be better to opt for the simple option here.

Reply:
@raxod502, I would still need to extract the recipe from the arguments of `use-package` to be able to call `straight-register-package`. What is the correct way to do this?

In the most recent push, I changed `use-feature` to `radian-use-feature` and created an alias.